NOTE the enum is not in  a class definition, and:
```
src/providers/postgres/qgspostgresconn.h:50:1: error: ‘friend’ used outside of class
   50 | Q_ENUM(QgsPostgresGeometryColumnType)
      | ^~~~~~
```

--- a/tests/src/providers/CMakeLists.txt
+++ b/tests/src/providers/CMakeLists.txt
@@ -71,6 +71,7 @@ target_link_libraries(qgis_postgresprovidertest postgresprovider_a qgis_core)
 
 ADD_QGIS_TEST(postgresconntest testqgspostgresconn.cpp)
 target_link_libraries(qgis_postgresconntest postgresprovider_a qgis_core)
+SET_TESTS_PROPERTIES(qgis_postgresconntest PROPERTIES LABELS "POSTGRES")
 
 if (NOT FORCE_STATIC_PROVIDERS)
   ADD_QGIS_TEST(mdalprovidertest testqgsmdalprovider.cpp)

